### PR TITLE
[Feature] Multi dub w/ season constraints

### PR DESCRIPTION
**Pull request by Denoder**
I created this because there are times where i want to get multiple dubs within a series but i cant do that with `-s` and `--srz` uses indexes rather than episode numbers for ordering. With this change the following output will be generated when using both `-s` and `--srz`:

```txt
[E1|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Failed... Hero [Japanese, English, Portuguese, Spanish]
[E2|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Black Spider of Disaster [Japanese, English, Portuguese, Spanish]
[E3|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Human Shock [Japanese, English, Portuguese, Spanish]
[E4|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Too Late [Japanese, English, Portuguese, Spanish]
[E5|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - The Greedy Wagon's Journey [Japanese, English, Portuguese, Spanish]
[E6|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - The Melancholy of Handsome Middle-Aged Men [Japanese, English, Portuguese, Spanish]
[E7|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Wondrous Medicine Production [Japanese, English, Portuguese, Spanish]
[E8|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Demiplane Ranking [Japanese, English, Portuguese, Spanish]
[E9|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Eat or Be Eaten [Japanese, Portuguese, Spanish, English]
[E10|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Hidden Ogre Village [Japanese, Portuguese, Spanish, English]
[E11|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Goodbye [Japanese, Portuguese, Spanish, English]
[E12|SGY2PCV21D] TSUKIMICHI -Moonlit Fantasy- - Season 1 - Guided by the Moon... [Japanese, Portuguese, Spanish, English]
[E1|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - What? Moon over the Ruined Castle? [☆ Japanese, ☆ English, ☆ Portuguese, ☆ Spanish]
[E2|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - The Heroes Are a Couple of Beauties [☆ Japanese, ☆ English, ☆ Spanish, ☆ Portuguese]
[E3|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - Stellar Wars [☆ Japanese, ☆ English, ☆ Portuguese, ☆ Spanish]
[E4|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - Why Am I a Teacher?! [☆ Japanese, ☆ English, ☆ Portuguese, ☆ Spanish]
[E5|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - The Lesson Will Now Begin [☆ Japanese, ☆ Portuguese, ☆ Spanish, ☆ English]
[E6|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - Becoming a Three-Star Chef [☆ Japanese, ☆ Spanish, ☆ Portuguese, ☆ English]
[E7|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - New Member Audition [☆ Japanese, ☆ English, ☆ Portuguese, ☆ Spanish]
[E8|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - The Notorious Beautiful Sisters [☆ Japanese, ☆ English, ☆ Portuguese, ☆ Spanish]
[E9|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - The Perverted Dragon [☆ Japanese, ☆ English]
[E10|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - Watch As I Improve the World [☆ Japanese]
[E11|SG69PC2JJ4] TSUKIMICHI -Moonlit Fantasy- Season 2 - Season 2 - Summer of Growth and New Skills [☆ Japanese]
```
With this, it sorts by season_id and episode (using season numbers is convoluted with cr's api)

if you exclude `-s` it falls back to the original format of using indexes (so it'll be number 1-23...etc) so overall this will retain the same functionality, but with the added effect that using `-s` will constrain it to the season and allow `-e` to use the episode number of that season.

### CODE REVIEW

**Comment by AnimeDL** [comment]
While the code looks good and appears to work, I do have a couple questions. 

1. When adding a season constraint by ID, wouldn't it be better not to show the other seasons in the first place? It may be confusing to see everything that isn't in the constraint. If that is done, then the Season ID next to the episode also becomes unnecessary. (It would also allow for `--all` for multi-dub on a specific season in a `--srz`)
2. When you say ordering, do you mean the order of the list specifically, or the resulting episode number ordering?

**Comment by Denoder** [comment]
1. Yes I just didn't change the logs to reflect that. I'll change it.
2. Episode numbering, like when im searching episode 11 in a series (while i was using `-s`) I was expecting to get episode 11 of season 2 but it gave me episode 11 of season 1 because it ignored the `-s` param.

**Comment by Denoder** [comment]
@AnimeDL I've made the changes

**Comment by AnimeDL** [comment]
Alright, did some more testing and this does seem to break the GUI in it's current state due to ./gui/server/services/crunchyroll.ts L22, since data isn't passed `data.s` doesn't exist and throws an exception. Once that's fixed, this is probably good to go

**Comment by Denoder** [comment]
Alright, I changed it.

One a side node if you're still looking to have the specials be located at the end of the episode listing (in the TODO comment) doing something like:

```ts
const specials = Object.entries(episodes).filter(a => a[0].startsWith('S')),
      normal = Object.entries(episodes).filter(a => a[0].startsWith('E')),
      sortedEpisodes = Object.fromEntries([...normal, ...specials]);
```

After they get prepended with `E` and `S` would work, then have then have the console.info for loop and the return object use `sortedEpisodes`

File: crunchy.ts [hunk 1]
```diff
@@ -2166,7 +2166,7 @@ export default class Crunchy implements ServiceClass {
       merger.cleanUp();
   }
 
-  public async listSeriesID(id: string): Promise<{ list: Episode[], data: Record<string, {
+  public async listSeriesID(id: string, data: CrunchyMultiDownload | undefined = undefined): Promise<{ list: Episode[], data: Record<string, {
```
Comment (AnimeDL) [inline crunchy.ts:2169]: Is there a reason to use `data: CrunchyMultiDownload | undefined = undefined` here instead of `data?: CrunchyMultiDownload`?

Comment (Denoder) [reply crunchy.ts]: changed, my vscode auto-suggests things so i don't usually argue with what it suggests.

**Comment by AnimeDL** [comment]
LGTM, Merging

**Pull request merged**